src/sb-util.ts

```diff
@@ -49,12 +39,26 @@ export class ScratchProject {
 		return new SpriteCollection(sprites).query(selector);
 	}
 
-	stage() {
+	stage(): Sprite {
 		return this.sprites('[isStage=true]').first();
 	}
 
 	blocks() {
-		throw new Error('ScratchProject blocks() function not implemented yet!');
+		const sprites = this.sprites();
+		const iters = [];
+		for(const sprite of sprites) {
```

Comment:
This must be done inside an iterator. This work must be done on demand, meaning after the BlockCollection is created.

Reply:
Specifically I'm talking about the for loop.

Reply:
I think you might turn

```js
makeIterable(iters, iters => chain(...iters));
```

into

```js
makeIterable(sprites, sprite => {
    return filter(sprite.blocks(), blocks => map(blocks, block => last(storage.get(block)));
});
```